docs/admin/multiple-schedulers.md

```diff
@@ -0,0 +1,214 @@
+---
+---
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Running Multiple Schedulers
+
+Kubernetes ships with a default scheduler that is described in [scheduler documentation](kube-scheduler.md). If the default scheduler shipped with Kubernetes does not suit your needs you can implement your own scheduler. Not just that, you can even run multiple schedulers simultaneously alongside the default scheduler and instruct Kubernetes what
+scheduler to use for each of your pods. Let's learn how to run multiple schedulers in Kubernetes with an example.
+
+Detailed description of how to implement a scheduler is outside the scope of this document. Please refer to the kube-scheduler implementation in [plugin/pkg/scheduler](https://github.com/kubernetes/kubernetes/tree/master/plugin/pkg/scheduler) in the Kubernetes source directory for a canonical example.
+
+### 1. Package the scheduler
+
+Package your scheduler binary into a container image. For the purposes of this example, let's just use kube-scheduler as our second scheduler as well. Clone the [Kubernetes source code from Github](https://github.com/kubernetes/kubernetes) and build the source.
+
+```sh
+$ git clone https://github.com/kubernetes/kubernetes.git
+$ cd kubernetes
+$ hack/build-go.sh
+```
+
+Copy the built kube-scheduler binary into the container image. Here is the `Dockerfile` to build the image:
+
+```Dockerfile
+FROM busybox
+ADD _output/local/go/bin/kube-scheduler /usr/local/bin/kube-scheduler
+```
+
+Save the file as `Dockerfile`, build the image and push it to a registry. This example pushes the image to [Google Container Registry (GCR)](https://cloud.google.com/container-registry/). For more details, please read the GCR [documentation](https://cloud.google.com/container-registry/docs/).
+
+```sh
+$ docker build -t my-kube-scheduler:1.0 .
+$ gcloud docker push gcr.io/my-gcp-project/my-kube-scheduler:1.0
+```
+
+### 2. Define a Kubernetes deployment for the scheduler
+
+Like most other system components, Kubernetes allows running a scheduler in a [pod](http://kubernetes.io/docs/user-guide/pods/). Since we have our scheduler in a container image already, we can just create a pod config for it and run it in a Kubernetes cluster. But instead of just running a pod in the cluster, let's use a [Deployment](http://kubernetes.io/docs/user-guide/deployments/) for this example. A [Deployment](http://kubernetes.io/docs/user-guide/deployments/) manages a ReplicaSet which in turn manages the pods, thereby making the scheduler resilient to failures. Here is the deployment config. Save it as `my-scheduler.yaml`:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    component: scheduler
+    tier: control-plane
+  name: my-scheduler
+  namespace: kube-system
+spec:
+  replicas: 1
+  selector:
+    matchExpressions:
+    - {key: tier, operator: In, values: [control-plane]}
```

Comment:
My goal was to introduce as many examples as possible with matchExpressions. There are very few examples right now. But I agree this is adding much value. So removing it.